File: NaumenScala/app/controllers/ShipController.scala
package controllers

import javax.inject._
import play.api.mvc._
import services._
import models._
import play.api.Logger


@Singleton
class ShipController @Inject()(cc : ControllerComponents, shipsFactoryService : ShipsFactoryService) extends AbstractController(cc) {

  def setupFactorySize() = Action { request: Request[AnyContent] =>
    try {
      val jsonBody = request.body.asJson
      val numberOfPlaces = jsonBody.get("numberOfPlaces").as[String].toInt
      shipsFactoryService.setupNewFactory(numberOfPlaces)
      Ok("")
    }
    catch {
      case _: Throwable => BadRequest("Bad Request")
    }
  }

  def addShip() = Action { request: Request[AnyContent] =>
    try {
      val jsonBody = request.body.asJson
      val timeOfArrival = jsonBody.get("timeOfArrival").as[String].toInt
      val handleTime = jsonBody.get("handleTime").as[String].toInt
      val ship = new Ship(timeOfArrival, handleTime)
      shipsFactoryService.addShipInQueue(ship)
      Ok("")
    }
    catch {
      case _: Throwable => BadRequest("Bad Request")
    }
  }

  def nextShipRepairTime() = Action { request: Request[AnyContent] =>

      val startTime = shipsFactoryService.nextRepairStartTime()
      if (startTime == null){
        Ok("")
      }
      else {
        Ok("{response: \"" + startTime.repairStartTime + "\"}")
      }
  }

  def methodNotFound(str: String): Action[AnyContent] = Action { request: Request[AnyContent] =>
    NotFound
  }
}

File: Naumen/CSharp/NaumenCSharp/Controllers/ShipController.cs
using System;
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Mvc;
using NaumenCSharp.Models;
using NaumenCSharp.Services;

namespace NaumenCSharp.Controllers
{
    [Route("")]
    [ApiController]
    public class ShipController : ControllerBase
    {
        private readonly ShipsFactoryService _factoryService;

        public ShipController(ShipsFactoryService factoryService)
        {
            _factoryService = factoryService;
        }

        [HttpPost("numberOfPlaces")]
        public ActionResult SetupFactorySize([Required] [FromBody] SetupNumberOfPlacesRequest request)
        {
            try
            {
                _factoryService.SetupNewFactory(request.NumberOfPlaces);
            }
            catch (Exception ex)
            {
                return BadRequest();
            }

            return Ok();
        }

        [HttpPost("ship")]
        public ActionResult AddShip([Required] [FromBody] Ship ship)
        {
            try
            {
                _factoryService.AddShipInQueue(ship);
            }
            catch (Exception ex)
            {
                return BadRequest();
            }

            return Ok();
        }

        [HttpGet("next")]
        public ActionResult GetNextShipRepairTime()
        {
            try
            {
                var repairEndTime = _factoryService.GetNextRepairStartTime();
                if (repairEndTime == null)
                    return Ok();

                return Ok(new GetNextShipResponse {Response = (int) repairEndTime});
            }
            catch (Exception ex)
            {
                return BadRequest();
            }
        }
    }
}